### PR TITLE
Chnage: switch to Webpack 4

### LINKED ISSUE #2930: Upgrade from webpack 3 to webpack 4
The title pretty much tells it all.

### PR DESCRIPTION
**Pull request by dmonego**
Hi! There are still a few warnings this is throwing, but I'd appreciate if project maintainers had any input on this changeover! 

Overall it seems like the default behavior of webpack 4 is pretty close to the custom chunks you were maintaining, so a fair amount of code gets saved in webpack.config.

Closes #2930.

### CODE REVIEW

**Comment by arikfr** [comment]
Amazing, thanks @dmonego ! 

It seems that the warnings about the size of bundle we create, but not about actual issues. Anyway, I will compare the generated files with what we currently have and then see if any changes are needed.

File: client/app/assets/less/main.less [hunk 1]
```diff
@@ -11,7 +11,6 @@
 @import '~ui-select/dist/select.css';
 @import '~angular-toastr/src/toastr';
 @import '~angular-resizable/src/angular-resizable.css';
-@import '~pace-progress/themes/blue/pace-theme-minimal.css';
```
Comment (arikfr) [inline client/app/assets/less/main.less]: Why?

Comment (dmonego) [reply client/app/assets/less/main.less]: As far as I can tell, it's a dead link.

Comment (arikfr) [reply client/app/assets/less/main.less]: It's not and it seems to be working for me with `MiniCssExtractPlugin`. I pushed a revert for this along with a few other fixes (see my commits in this PR). Once I see the build works -- will merge. 👍

**Comment by arikfr** [comment]
Merged. Thanks!

**Pull request merged**